crates/gossip/src/state.rs

```diff
@@ -0,0 +1,194 @@
+use crate::protocol::{
+    K2GossipAcceptMessage, K2GossipAgentsMessage, K2GossipNoDiffMessage,
+};
+use kitsune2_api::{AgentId, K2Error, K2Result, Url};
+use rand::RngCore;
+
+/// The state of a gossip round.
+#[derive(Debug)]
+pub(crate) struct GossipRoundState {
+    /// The agent id of the other party who is participating in this round.
+    pub session_with_peer: Url,
+
+    /// The time at which this round was initiated.
+    ///
+    /// This is used to apply a timeout to the round.
+    #[allow(dead_code)]
+    started_at: std::time::Instant,
+
+    /// The session id of this round.
+    ///
+    /// Must be randomly chosen and unique for each initiated round.
+    pub session_id: bytes::Bytes,
+
+    /// The current stage of the round.
+    ///
+    /// Store the current stage, so that the next stage can be validated.
+    pub stage: RoundStage,
+}
+
+impl GossipRoundState {
+    /// Create a new gossip round state.
+    pub(crate) fn new(
+        session_with_peer: Url,
+        our_agents: Vec<AgentId>,
+    ) -> Self {
+        let mut session_id = bytes::BytesMut::with_capacity(96);
+        rand::thread_rng().fill_bytes(&mut session_id);
```

Comment:
I don't think this is working. The `with_capacity` will still have a `len()` of zero, so the slice here will be a zero-length slice that doesn't get any random bytes in it.

Reply:
Always get this wrong with bytes... Fixed and added a unit test